test/integration/docker_test.go

```diff
@@ -158,3 +164,114 @@ func TestForceSystemdEnv(t *testing.T) {
 		validateContainerdSystemd(ctx, t, profile)
 	}
 }
+
+// TestDockerEnvContainerd makes sure that minikube docker-env command works when the runtime is containerd
+func TestDockerEnvContainerd(t *testing.T) {
```

Comment:
@ComradeProgrammer I am curious what was the reason of creating a brand new test just for this 

could we use this test instead ?
(validateDockerEnv) which is in "functional_test.go" ?
https://github.com/medyagh/minikube/blob/0610a1f81e5510c7982be27a922039e260c2140f/test/integration/functional_test.go#L451

in there we could add to the if statement
```
	// docs(skip): Skips on non-docker container runtime
	if cr := ContainerRuntime(); cr == "crio" {
		t.Skipf("only validate docker env with docker and containerd runtimes, currently testing %s", cr)
	}
```

Reply:
I think perhaps this validateDockerEnv test is too different from the unit test in this PR.
 
First, there is a lot of configurations we need to do for the containerd docker-env test, like `ssh-agent bash`, banning DOCKER_BUILDKIT....

 Second, the things we need to check in the test is also too different. We need to check the existence of nerdctld, check whether it can build a image, which is not a part of  validateDockerEnv. 

Besides, I think this test should be seperated from the functional tests, because to perform this test we changed so many configurations like ssa keys, envs. If we write this test in functional tests and if something wrong (like a panic) happens in this test, it may influence the following tests. I think it is better to start a new minikube cluster for this test, and destroy this cluster after this test.